cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,339 @@
+    )
+
+    paginator = client.get_paginator("lookup_events")
+
+    page_iterator = paginator.paginate(
+        LookupAttributes=[
+            {"AttributeKey": "EventName", "AttributeValue": "AssumeRole"}
+        ],
+        StartTime=start_time,
+        EndTime=end_time,
+        PaginationConfig={
+            "MaxItems": 10000,  # Reasonable limit to prevent excessive API calls
+            "PageSize": 50,  # CloudTrail API limit per page
+        },
+    )
+
+    all_events = []
+    for page in page_iterator:
+        all_events.extend(page.get("Events", []))
+
+    logger.info(f"Retrieved {len(all_events)} AssumeRole events from region '{region}'")
+
+    return all_events
+
+
+@timeit
+def transform_assume_role_events_to_role_assumptions(
+    events: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+) -> List[Dict[str, Any]]:
+    """
+    Transform raw CloudTrail AssumeRole events into aggregated role assumption relationships.
+
+    Focuses specifically on standard AssumeRole events, providing optimized processing
+    for the most common role assumption scenario.
+
+    This function performs the complete transformation pipeline:
+    1. Extract role assumption events from CloudTrail AssumeRole data
+    2. Aggregate events by (source_principal, destination_principal) pairs
+    3. Return aggregated relationships ready for loading
+
+    :type events: List[Dict[str, Any]]
+    :param events: List of raw CloudTrail AssumeRole events from lookup_events API
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: List[Dict[str, Any]]
+    :return: List of aggregated role assumption relationships ready for loading
+    """
+    aggregated: Dict[tuple, Dict[str, Any]] = {}
+    logger.info(
+        f"Transforming {len(events)} CloudTrail AssumeRole events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+
+        # Extract source and destination principals
+        cloudtrail_event = json.loads(event["CloudTrailEvent"])
+
+        # Extract source principal with error handling (UserIdentity may be missing in some events)
+        try:
+            source_principal = cloudtrail_event["userIdentity"]["arn"]
+        except KeyError:
+            logger.debug(
+                f"Skipping CloudTrail AssumeRole event due to missing UserIdentity.arn. Event: {event.get('EventId', 'unknown')}"
+            )
+            continue
+
+        destination_principal = cloudtrail_event["requestParameters"]["roleArn"]
+
+        normalized_source_principal = _convert_assumed_role_arn_to_role_arn(
+            source_principal
+        )
+        normalized_destination_principal = _convert_assumed_role_arn_to_role_arn(
+            destination_principal
+        )
+        event_time = event.get("EventTime")
```

Comment:
[block] event time can be None. If this happens then the `<` and `>` checks below will crash.

i would do  somethign like

```python
if not event_time:
    aggregated[key]["first_seen_in_time_window"] = None
    aggregated[key]["last_used"] = None
else:
    aggregated[key]["first_seen_in_time_window"] = min(aggregated[key]["first_seen_in_time_window"], event_time)
    aggregated[key]["last_used"] = max(aggregated[key]["last_used"], event_time)
```

Note also that the use of min and max above helps consolidate a little bit, but that part of suggestion is optional